lib/modules/manager/gomod/artifacts.ts

```diff
@@ -243,7 +243,7 @@ export async function updateArtifacts({
       }
     }
 
-    let args = `get -d -t ${goGetDirs ?? './...'}`;
+    let args = `get -t ${goGetDirs ?? './...'}`;
```

Comment:
extract go version from `go` and / or `toolchain` directive. drop `-d` only if it is `>=1.18`.

Reply:
Sorry was just about to comment: I made an update to update Go version after I realised it was currently older: https://github.com/renovatebot/renovate/pull/32514